src/core/crypto/hdKey.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { hmac } from "@noble/hashes/hmac";
+import { sha512 } from "@noble/hashes/sha512";
+import * as bip39 from "@scure/bip39";
+
+export type DerivedKeys = {
+  key: Uint8Array;
+  chainCode: Uint8Array;
+};
+
+/**
+ * Aptos derive path is 637
+ */
+export const APTOS_HARDENED_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'?$/;
+export const APTOS_BIP32_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+\/[0-9]+$/;
+
+/**
+ * A list of supported key types and associated seeds
+ */
+export enum KeyType {
```

Comment:
Where is secp256k1? It should be here set to "Bitcoin seed", no?

Reply:
But actually, do you even use `KeyType` anywhere anymore?

Reply:
no. we use `const { privateKey } = HDKey.fromMasterSeed(mnemonicToSeed(mnemonics)).derive(path)`
https://github.com/aptos-labs/aptos-ts-sdk/pull/146/files#diff-98942ba717d1ba7eafd0b4cfa834f6307d611d3ba525aa2ae68a93938b97e0b5R189

Reply:
So, then remove this enum?

Reply:
cuz we need it for the Ed25519? and maybe for future implementation? tbh, it was implemented here before

Reply:
I'm confused. I thought you just said it's not used anywhere...

Reply:
oh no, I meant we are not using it for secp256k1

Reply:
Hm... I see, it might be confusing to have this incomplete enum lying around...